crates/bevy_ecs/src/system/query.rs

```diff
@@ -693,11 +736,32 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
     /// }
     /// # bevy_ecs::system::assert_is_system(system);
     /// ```
+    /// # See also
+    ///
+    /// - [`iter_many`](Self::iter_many) to get read-only query items.
+    /// - [`iter_many_inner`](Self::iter_many_inner) to get mutable query items with the full `'world` lifetime.
     #[inline]
     pub fn iter_many_mut<EntityList: IntoIterator<Item: EntityBorrow>>(
         &mut self,
         entities: EntityList,
     ) -> QueryManyIter<'_, 's, D, F, EntityList::IntoIter> {
+        self.reborrow().iter_many_inner(entities)
+    }
+
+    /// Returns an iterator over the query items generated from an [`Entity`] list.
```

Comment:
Should these `iter_many_inner`, `iter_combinations_inner` etc. call out in their docs why they are separate methods like `get_inner` does ("with the actual "inner" world lifetime, by consuming the [`Query`]")? To be clear, I'm not sure if they should but thought I should raise it as a possibility.